### ISSUE TITLE
IndexOutOfBoundsException in HabitCardListCache$RefreshTask.performMove

### ISSUE THREAD

**Issue by iSoron** [iSoron/uhabits#968]
## Pre-submission checklist
- [x] I am submitting a bug report, not a feature request.
- [x] I have searched for similar issues, but did not find any matches.

## Description
From Google Play Console:
```
java.lang.IndexOutOfBoundsException: 
  at java.util.LinkedList.checkPositionIndex (LinkedList.java:561)
  at java.util.LinkedList.add (LinkedList.java:508)
  at org.isoron.uhabits.core.ui.screens.habits.list.HabitCardListCache$RefreshTask.performMove (HabitCardListCache.kt:319)
  at org.isoron.uhabits.core.ui.screens.habits.list.HabitCardListCache$RefreshTask.processPosition (HabitCardListCache.kt:346)
  at org.isoron.uhabits.core.ui.screens.habits.list.HabitCardListCache$RefreshTask.onProgressUpdate (HabitCardListCache.kt:299)
  at org.isoron.uhabits.tasks.AndroidTaskRunner$CustomAsyncTask.onProgressUpdate (AndroidTaskRunner.kt:86)
  at org.isoron.uhabits.tasks.AndroidTaskRunner$CustomAsyncTask.<init> (AndroidTaskRunner.kt:57)
  at org.isoron.uhabits.tasks.AndroidTaskRunner$CustomAsyncTask.onProgressUpdate (AndroidTaskRunner.kt:57)
  at android.os.AsyncTask$InternalHandler.handleMessage (AsyncTask.java:716)
  at android.os.Handler.dispatchMessage (Handler.java:112)
  at android.os.Looper.loop (Looper.java:216)
  at android.app.ActivityThread.main (ActivityThread.java:7625)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:524)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:987)
```

## Steps to reproduce
Unknown

## System information
 - Phone: Multiple (Samsung, Huawei)
 - Phone Operating System: Multiple (Android 9, 11)
 - App version: 2.0.2

**Comment by hiqua** [comment]
I don't know the exact origin of this bug, but for now, would it be enough to change `toPosition` in this statement:
```kt
        @Synchronized
        private fun performMove(
            habit: Habit,
            fromPosition: Int,
            toPosition: Int
        ) {
            data.habits.removeAt(fromPosition)
            data.habits.size
            data.habits.add(toPosition, habit)
            listener.onItemMoved(fromPosition, toPosition)
        }
```
to something like `min(toPosition, data.habits.size)`? And we could log a warning with the change of indexes.

**Comment by iSoron** [comment]
@hiqua Yes, I think that would be fine.

**Comment by iSoron** [comment]
Fixed in branch `hotfix/2.0.3`
